bundler/lib/bundler/rubygems_ext.rb

```diff
@@ -266,23 +266,27 @@ def ===(other)
   end
 
   Platform.singleton_class.module_eval do
-    unless Platform.singleton_methods.include?(:match_spec?)
-      def match_spec?(spec)
-        match_gem?(spec.platform, spec.name)
-      end
+    if Platform.singleton_methods.include?(:match_spec?)
+      remove_method :match_spec?
+      remove_method :match_gem?
+      remove_method :match_platforms?
+    end
 
-      def match_gem?(platform, gem_name)
-        match_platforms?(platform, Gem.platforms)
-      end
+    def match_spec?(spec)
+      match_gem?(new(spec.platform), spec.name)
+    end
 
-      private
+    def match_gem?(platform, gem_name)
+      match_platforms?(platform, Gem.platforms)
+    end
 
-      def match_platforms?(platform, platforms)
-        platforms.any? do |local_platform|
-          platform.nil? ||
-            local_platform == platform ||
-            (local_platform != Gem::Platform::RUBY && local_platform =~ platform)
-        end
+    private
+
+    def match_platforms?(platform, platforms)
+      platforms.any? do |local_platform|
+        platform.nil? ||
+          local_platform == platform ||
```

Comment:
Would be good to specify `platform` must be a Gem::Platform as a comment or so.
I guess the problematic cases of `platform` being a String are `local_platform == platform`  and `platform =~ local_platform`, right?

Reply:
Yes, it's `platform =~ local_platform`, which calls `Gem::Platform#===` under the hood which requires other platforms to ever return `true`.